src/test/java/seedu/address/logic/commands/EditCommandTest.java

```diff
@@ -34,18 +37,40 @@
  */
 public class EditCommandTest {
 
-    private Model model = new ModelManager(getTypicalPennyWise(), new UserPrefs());
+    private Model model;
+
+    @BeforeEach
+    public void setModel() {
+        model = new ModelManager(getTypicalPennyWise(), new UserPrefs());
+    }
 
     @Test
     public void execute_allFieldsSpecifiedUnfilteredList_success() {
-        Entry editedEntry = new ExpenditureBuilder().withDescription("Acai").build();
-        EditEntryDescriptor descriptor = new EditEntryDescriptorBuilder(editedEntry, new EntryType("e")).build();
-        EditCommand editCommand = new EditCommand(INDEX_FIRST_PERSON, descriptor);
+        Index indexLastEntry = Index.fromOneBased(model.getFilteredExpenditureList().size());
+        Entry lastEntry = model.getFilteredExpenditureList().get(indexLastEntry.getZeroBased());
+
+        ExpenditureBuilder expenditureInList = new ExpenditureBuilder(lastEntry);
+
+        Entry editedEntry = expenditureInList
+                .withDescription(VALID_DESC_TUITION)
+                .withAmount(VALID_AMT_TUITION)
+                .withTag(VALID_TAG_PERSONAL)
+                .withDate(VALID_DATE_TUITION)
+                .build();
+        // Specify all fields to be edited
```

Comment:
Based on the test description, it seems more appropriate that we specify all fields for the `EditEntryDescriptor` instead of just `description` and `entryType` as before.